src/interpolators.js

```diff
@@ -27,7 +29,48 @@ export function isTemplateVariable(str) {
  * Interpolates a string, replacing template variables with values
  * provided in the scope.
  *
- * Besides replacing variables with
+ * @param  {String}    str    A string to be interpolated
+ * @param  {Object}    scope  An object with variable names and their
+ *                            replacements
+ * @return {String}           An interpolated string
+ */
+export function interpolateString(str, scope = {}) {
+  if (!str) {
+    return str
+  }
+
+  // Split string into array with regular text and variables split
+  // into separate segments, like ['This is a ', '{{ thing }}', '!']
+  const parts = str.split(new RegExp(variableRegex)).filter(x => x)
+
+  // If the only thing we have is a single regular string, just return it as is
+  if (parts.length === 1 && isTemplateVariable(parts[0]) === false) {
+    return str
+  }
+
+  return parts
+    .map(part => {
+      if (isTemplateVariable(part) === false) {
+        return part
+      }
+
+      const variableName = part.replace(/^\{\{\s/, '').replace(/\s\}\}$/, '')
+      if (scope[variableName] === undefined) {
```

Comment:
Should we generate a console warning here? As I would imagine this may be a bug?

Reply:
If so, it needs to be under a `debug` flag. I think it's fairly easy to debug and find these issues yourself, and am happy to leave it as it is.